src/snowflake/snowpark/_internal/compiler/utils.py

```diff
@@ -323,3 +327,75 @@ def get_snowflake_plan_queries(
 def is_active_transaction(session):
     """Check is the session has an active transaction."""
     return session._run_query("SELECT CURRENT_TRANSACTION()")[0][0] is not None
+
+
+def plot_plan_if_enabled(root: TreeNode, filename: str) -> None:
+    """A helper function to plot the query plan tree using graphviz useful for debugging.
+    It plots the plan if the environment variable ENABLE_SNOWFLAKE_OPTIMIZATION_PLAN_PLOTTING
+    is set to true.
+
+    The plots are saved in the temp directory of the system which is obtained using
+    https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir. Setting env variable
+    TMPDIR to your desired location is recommended. Within the temp directory, the plots are
+    saved in the directory `snowpark_query_plan_plots` with the given `filename`.
+
+    Args:
+        root: root TreeNode of the plan to plot.
+        filename: name of the file to save the image plot.
+    """
+    import os
+
+    import graphviz  # pyright: ignore[reportMissingImports]
+
+    if (
+        os.environ.get("ENABLE_SNOWFLAKE_OPTIMIZATION_PLAN_PLOTTING", "false").lower()
```

Comment:
maybe just call this ENABLE_SNOWPARK_LOGICAL_PLAN_PLOTTING since that is a snwopark plan not a snowflake plan